lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -791,7 +791,14 @@ export class UserAgentApplication {
         this.logger.verbose("Set loading state to pending for: " + scope + ":" + expectedState);
         this.cacheStorage.setItem(`${TemporaryCacheKeys.RENEW_STATUS}${Constants.resourceDelimiter}${expectedState}`, Constants.inProgress);
 
-        const iframe = await WindowUtils.loadFrame(urlNavigate, frameName, this.config.system.navigateFrameWait, this.logger);
+        // render the iframe synchronously if app chooses no timeout, else wait for the set timer to expire
+        let iframe: HTMLIFrameElement;
+        if(this.config.system.navigateFrameWait == 0) {
+            iframe = WindowUtils.syncLoadFrame(urlNavigate, frameName, this.logger);
+        }
+        else {
+            iframe = await WindowUtils.loadFrame(urlNavigate, frameName, this.config.system.navigateFrameWait, this.logger);
+        }
```

Comment:
A couple things:

- You can simplify the check to test truthiness of `navigateFrameWait`, since `0` is falsey.
- Can you make this a ternary?

```js
const iframe = this.config.system.navigateFrameWait ? 
  await WindowUtils.loadFrame(urlNavigate, frameName, this.config.system.navigateFrameWait, this.logger) : 
  WindowUtils.syncLoadFrame(urlNavigate, frameName, this.logger);
```

Reply:
I thought of it but wanted to test it before making it ternary as numbers and strings behave differently in TS :-D. Will do.

Reply:
+1 to ternary

Reply:
Done!